Complex SIR model WITH variability.Rmd

```diff
@@ -0,0 +1,167 @@
+---
+title: "Complex SIR model with Variability"
+author: "Linh, Britney, Bowen"
+output: html_notebook
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(warning = FALSE, message = FALSE)
+library(mosaic)
+library(ggplot2)
+```
+
+## Complex SIR Model
+
+### Assumption
+
+* The recovered (R) compartment will no longer be susceptible
+
+### Parameters with default values:
+
+* Population $(N = 300)$
+
+* Transmission/Infection Rate $(r = 0.01)$
+
+* Percentage of healthy individuals with symptoms $(g = 0.2)$
+
+* Initial number of infected individuals day1 $(d_1 = 5)$
+
+* Initial number of cured individuals day1 $(c_1 = 0)$
+
+* Percentage of individuals receiving Treatment A $(PTA = 1)$
+
+* Percentage of individuals receiving Treatment B $(PTB = 0)$
+
+* Treatment A effectiveness $(A_{eff} = 0.75)$
+
+* Treatment B effectiveness $(B_{eff} = 0.25)$
+
+* Treatment A cost $(A_{cost} = 20.00)$
+
+* Treatment B cost $(B_{cost} = 15.00)$
+
+* Number of days since first case $(day = 38)$
+
+*Model upgrade with randomness - Additional Variables*
+
+* Initial number of known cases (untreated) day1 $(u_1 = 0)$
+
+* Percentage of individual being tested daily $(PS = 1)$
+
+* Test sensitivity $(tp = 0.99)$
+
+* Test specificity $(tn = 0.95)$
+
+* Testing cost $(Test.cost = 1)$
+
+```{r model.rand}
+complex.model.rand = function(N=300, r=0.01, g=0.2, d1=5, c1=0, u1=0, PS=1, tp=0.99, tn=0.95, PTA=1, PTB=0, A.eff=0.75, B.eff=0.25, day = -1, A.cost = 20, B.cost = 15, Test.cost = 1){
+  
+  # Initial conditions
+  curr.day = 1
+  d = d1
+  c = c1
+  u = u1
+  # Calculate initial statistics
+  h = N - d - c
+  hs = round(g * h)
+  s = hs + d
+  a = s - u
+  ns = round(PS * a)
+  ad = round(PS * (d - u))
+  as = ns - ad
+  tps = rbinom(1, as, 1-tn)
+  tpd = rbinom(1, ad, tp)
+  tpt = tps + tpd
+  tpt1 = tpt
+  NDAT = tpd + u
+  NDTA = 0
+  NDTB = 0
+  DCA = 0
+  DCB = 0
+  NAT = 0
+  NTA = 0
+  NTB = 0
+  SDC = 0
+  New = min(N - c - d, round(r * hs * d))
```

Comment:
Do you know why this is the function for people who catch the disease? I am a little confused.